src/framework/net/httplogin.cpp

```diff
@@ -136,6 +142,63 @@ void LoginHttp::httpLogin(const std::string& host, const std::string& path,
             });
         }
     });
+#else
+    g_asyncDispatcher.detach_task(
+        [this, host, path, port, email, password, request_id, httpLogin] {
+        emscripten_fetch_attr_t attr;
+        emscripten_fetch_attr_init(&attr);
+        strcpy(attr.requestMethod, "POST");
+        static const char* const headers[] = {
+            "Content-Type", "application/json; charset=utf-8",
+            0,
+        };
+        attr.requestHeaders = headers;
+        attr.attributes = EMSCRIPTEN_FETCH_LOAD_TO_MEMORY | EMSCRIPTEN_FETCH_SYNCHRONOUS;
+        std::string body = "{\"email\":\"" + email + "\",\"password\":\"" + password + "\",\"stayloggedin\":true,\"type\":\"login\"}";
+        attr.requestData = body.c_str();
+        attr.requestDataSize = strlen(attr.requestData);
+
+        std::string url = "https://" + (host.length() > 0 ? host : "127.0.0.1") + ":" + std::to_string(port) + path;
+        emscripten_fetch_t* fetch = emscripten_fetch(&attr, url.c_str());
+
+        if (fetch->status != 200 && httpLogin) {
+            std::string url = "http://" + (host.length() > 0 ? host : "127.0.0.1") + ":" + std::to_string(port) + path;
+            fetch = emscripten_fetch(&attr, url.c_str());
+        }
+
+        if (fetch && fetch->status == 200 &&
+               !parseJsonResponse(std::string(fetch->data, fetch->numBytes))) {
+            fetch->status = -1;
+        }
```

Comment:
I think you meant
```suggestion
        if (!fetch || fetch->status != 200 ||
               !parseJsonResponse(std::string(fetch->data, fetch->numBytes))) {
            fetch->status = -1;
        }
```
?